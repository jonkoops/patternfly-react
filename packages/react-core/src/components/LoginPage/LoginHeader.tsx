import { HTMLProps, ReactNode, FunctionComponent } from 'react';
import styles from '@patternfly/react-styles/css/components/Login/login';
import { css } from '@patternfly/react-styles';

export interface LoginHeaderProps extends HTMLProps<HTMLDivElement> {
  /** Content rendered inside the header of the login layout */
  children?: ReactNode;
  /** Additional classes added to the login header */
  className?: string;
  /** Header brand component (e.g. <LoginHeader />) */
  headerBrand?: ReactNode;
}

export const LoginHeader: FunctionComponent<LoginHeaderProps> = ({
  className = '',
  children = null,
  headerBrand = null,
  ...props
}: LoginHeaderProps) => (
  <header className={css(styles.loginHeader, className)} {...props}>
    {headerBrand}
    {children}
  </header>
);
LoginHeader.displayName = 'LoginHeader';
