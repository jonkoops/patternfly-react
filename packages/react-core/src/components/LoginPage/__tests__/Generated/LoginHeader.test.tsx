/**
 * @vitest-environment jsdom
 */
import { render } from '@testing-library/react';
import React from 'react';
import { expect, it } from 'vitest';

import { LoginHeader } from '../../LoginHeader';

it('LoginHeader should match snapshot (auto-generated)', () => {
  const { asFragment } = render(<LoginHeader children={<>ReactNode</>} className={"''"} headerBrand={null} />);
  expect(asFragment()).toMatchSnapshot();
});
