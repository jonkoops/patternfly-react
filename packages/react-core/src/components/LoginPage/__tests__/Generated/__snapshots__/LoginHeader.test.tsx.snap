// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoginHeader should match snapshot (auto-generated) 1`] = `
<DocumentFragment>
  <header
    class="pf-v5-c-login__header ''"
  >
    ReactNode
  </header>
</DocumentFragment>
`;
